modin/core/dataframe/pandas/dataframe/utils.py

```diff
@@ -71,9 +71,11 @@ def pivot_fn(samples):
             samples, ideal_num_new_partitions, method, key
         )
 
+    dtypes = modin_frame.dtypes
+
     def split_fn(partition, pivots):
         return split_partitions_using_pivots_for_sort(
-            modin_frame, partition, column, pivots, ascending, **kwargs
```

Comment:
We cannot use this object in a remote context, because it contains methods which use the functionality of engines (for example, `_compute_dtypes()`).